tests/util/base.py

```diff
@@ -0,0 +1,117 @@
+        TcFileTarget.__init__(self, iface, direction=direction)
+        self._callback = callback
+
+    def install(self, verbose=False):
+        self._callback(self._commands)
+
+
+class LtcSimulateTargetRun(object):
+    """Used by simulation pyltc tests.
+    Executes given pyltc command using the TcTestTarget."""
+
+    def __init__(self, argv, full=False):
+        self._argv = argv
+        self._full = full
+        self._result = None
+
+    def our_callback(self, result):
+        self._result = result
+
+    def test_target_factory(self, iface, direction):
+        return TcTestTarget(iface, self.our_callback, direction)
+
+    def run(self):
+        pyltc_entry_point(self._argv, self.test_target_factory)
+
+    @property
+    def result(self):
+        return self._result
+
+
+class LtcLiveTargetRun(object):
+    """Used by live pyltc tests.
+    Executes given pyltc command using TcCommandTarget
+    and measures rates, then returns results."""
+
+    def __init__(self, argv, udp_sendrate, duration=5, full=False):
+        self._argv = argv
+        self._udp_sendrate = udp_sendrate
+        self._duration = duration
+        self._full = full
+        self._result = None
+
+    def _target_factory(self, iface, direction):
+        return TcCommandTarget(iface, direction)
+
+    def _test_for_portrange(self, klass, port_range, sendrate):
+        if '-' in port_range:
+            left_in_port, right_in_port = map(int, port_range.split('-'))
+            left_out_port = left_in_port - 1
+            right_out_port = right_in_port + 1
+
+        else:
+            left_in_port = right_in_port = int(port_range)
+            left_out_port = left_in_port - 1
+            right_out_port = right_in_port + 1
+
+        bandwidth_dict = {}
+        tcp_netperf_1 = klass(sendrate, ip='127.0.0.1', port=left_in_port, duration=self._duration)
+        bandwidth_dict['left_in'] = tcp_netperf_1.run()
+        tcp_netperf_2 = klass(sendrate, ip='127.0.0.1', port=left_out_port, duration=self._duration)
+        bandwidth_dict['left_out'] = tcp_netperf_2.run()
+        tcp_netperf_3 = klass(sendrate, ip='127.0.0.1', port=right_in_port, duration=self._duration)
+        bandwidth_dict['right_in'] = tcp_netperf_3.run()
+        tcp_netperf_4 = klass(sendrate, ip='127.0.0.1', port=right_out_port, duration=self._duration)
+        bandwidth_dict['right_out'] = tcp_netperf_4.run()
+
+        return bandwidth_dict
+
+    def run(self):
+        orig_argv = self._argv[:]
+        pyltc_entry_point(self._argv, self._target_factory)
+        args = parse_args(orig_argv)
+        result = {}
+        for dc in args.dclass:
+            dc_dict = parse_branch(dc)
+            klass = TCPNetPerfTest if dc_dict['protocol'] == 'tcp' else UDPNetPerfTest
+            bandwidth_dict = self._test_for_portrange(klass, dc_dict['range'], self._udp_sendrate)
+            result[dc] = bandwidth_dict
+
+        self._result = result
+#         return {
```

Comment:
This comment shows what the function returns (since it is not too simple); as of now the random ports are not yet implemented, but it also works as a reminder for that.
Will add explanation in the comment.